lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -1754,6 +1766,30 @@ Activity DifferentiableActivityInfo::getActivity(
   return activity;
 }
 
+void DifferentiableActivityInfo::collectAllActiveValues(
+    SILBasicBlock *bb, const SILAutoDiffIndices &indices,
+    SmallVectorImpl<SILValue> &activeValues) const {
+  SmallPtrSet<SILValue, 8> registered;
+  // Register a value as active if it has not yet been registered.
+  auto addActiveValue = [&](SILValue v) {
+    if (registered.count(v))
+      return;
+    registered.insert(v);
+    activeValues.push_back(v);
+  };
+  for (auto *arg : bb->getArguments())
+    if (isActive(arg, indices))
+      addActiveValue(arg);
+  for (auto &inst : *bb) {
+    for (auto op : inst.getOperandValues())
```

Comment:
If you iterate over each instruction's operands, then it's not just collecting active values that are defined in the basic block, but also all active values that are used in the BB. Is that the intended behavior?

Reply:
Yes, that is intended behavior.

If `%0` is an argument of `bb0` but is used in `bb1`, then `bb1_adj` (adjoint block of bb1) needs `%0_adj` as a BB argument.